src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -1355,45 +1265,91 @@ public void doUpdateLoadFlowParametersAndStatus(UUID studyUuid, LoadFlowParamete
         });
     }
 
-    public Flux<ModificationInfos> getModifications(UUID studyUuid) {
-        return getNetworkUuid(studyUuid)
-                .flatMapMany(networkUuid -> {
-                    var path = UriComponentsBuilder.fromPath(DELIMITER + NETWORK_MODIFICATION_API_VERSION + "/networks/{networkUuid}/modifications")
-                            .buildAndExpand(networkUuid)
-                            .toUriString();
-                    return webClient.get().uri(networkModificationServerBaseUri + path).retrieve().bodyToFlux(new ParameterizedTypeReference<ModificationInfos>() {
-                    });
+    private void sendUpdateMessage(Message<String> message) {
+        MESSAGE_OUTPUT_LOGGER.debug("Sending message : {}", message);
+        studyUpdatePublisher.send("publishStudyUpdate-out-0", message);
+    }
+
+    Mono<List<String>> getAvailableSvgComponentLibraries() {
+        String path = UriComponentsBuilder.fromPath(DELIMITER + SINGLE_LINE_DIAGRAM_API_VERSION + "/svg-component-libraries")
+                .toUriString();
+
+        return webClient.get()
+                .uri(singleLineDiagramServerBaseUri + path)
+                .retrieve()
+                .bodyToMono(new ParameterizedTypeReference<>() {
                 });
     }
 
+    @Transactional
+    public void doUpdateIndexingStatus(UUID studyUuid, IndexingStatus indexingStatus) {
+        studyRepository.findById(studyUuid).ifPresent(studyEntity -> studyEntity.setIndexingStatus(indexingStatus));
+    }
+
+    Mono<Void> updateIndexingStatus(UUID studyUuid, IndexingStatus indexingStatus) {
+        return Mono.fromRunnable(() -> self.doUpdateIndexingStatus(studyUuid, indexingStatus));
+    }
+
+    public Flux<ModificationInfos> getModifications(UUID studyUuid) {
+        Objects.requireNonNull(studyUuid);
```

Comment:
Not coherent
why we don't use getGroupUuid ?
STUDY_NOT_FOUND not throwed

Reply:
Done